lms/djangoapps/instructor/views/api.py

```diff
@@ -231,6 +233,28 @@ def wrapped(*args, **kwargs):
     return decorator
 
 
+def verify_course_permission(permission):
```

Comment:
I have added new decorator because in existing one they are not passing `self` in constructor.

Reply:
Instead of making a new decorator, does it make more sense to make a custom permission class instead?